android/databases/src/main/java/com/coinbase/wallet/libraries/databases/db/Database.kt

```diff
@@ -0,0 +1,264 @@
+package com.coinbase.wallet.libraries.databases.db
+
+import androidx.room.Room
+import androidx.sqlite.db.SimpleSQLiteQuery
+import com.coinbase.wallet.libraries.databases.exceptions.DatabaseException
+import com.coinbase.wallet.libraries.databases.interfaces.DatabaseDaoInterface
+import com.coinbase.wallet.libraries.databases.interfaces.DatabaseModelObject
+import com.coinbase.wallet.libraries.databases.model.DiskOptions
+import com.coinbase.wallet.libraries.databases.model.MemoryOptions
+import com.gojuno.koptional.Optional
+
+import io.reactivex.Observable
+import io.reactivex.Single
+import io.reactivex.schedulers.Schedulers
+import io.reactivex.subjects.PublishSubject
+import java.util.concurrent.ConcurrentHashMap
+
+class Database<T : RoomDatabaseProvider>() {
+    private lateinit var provider: RoomDatabaseProvider
+
+    constructor(disk: DiskOptions<T>) : this() {
+        provider = Room.databaseBuilder(disk.context, disk.providerClazz, disk.dbName).build()
```

Comment:
One thing missing from this class is the ability to add migrations to datbases.  In room, you typically specify a default (fallback) migration.  Then you can add arbitrary per-version migrations.  Without changing too much, we can add a list of migrations to the `DiskOptions` class, and add a boolean (default true) whether it should fallback to destructive migration or not.

Reply:
done